src/views/XBlockStudioView/components/StudioViewSettings/FileUploadConfig.test.jsx

```diff
@@ -1,34 +1,74 @@
-import { shallow } from '@edx/react-unit-test-utils';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
 
 import { useFileUploadConfig } from 'hooks/app';
 
 import FileUploadConfig from './FileUploadConfig';
 
+jest.unmock('@openedx/paragon');
+jest.unmock('react');
+jest.unmock('@edx/frontend-platform/i18n');
+
 jest.mock('hooks/app', () => ({
   useFileUploadConfig: jest.fn(),
 }));
-jest.mock('./RequiredConfig', () => 'RequiredConfig');
+jest.mock('./RequiredConfig', () => {
+  // eslint-disable-next-line react/prop-types
+  const RequiredConfig = ({ required }) => (
+    <span data-testid="required-config">
+      {required ? 'Required' : 'Optional'}
+    </span>
+  );
+  return RequiredConfig;
+});
 
 describe('<FileUploadConfig />', () => {
-  it('should render', () => {
+  const renderWithIntl = (component) => render(<IntlProvider locale="en">{component}</IntlProvider>);
+
+  it('renders file upload configuration when enabled', () => {
     useFileUploadConfig.mockReturnValue({
       enabled: true,
       fileUploadLimit: 10,
       required: true,
     });
 
-    const wrapper = shallow(<FileUploadConfig />);
-    expect(wrapper.snapshot).toMatchSnapshot();
+    renderWithIntl(<FileUploadConfig />);
+
+    expect(screen.getByText('File uploads:')).toBeInTheDocument();
+    expect(screen.getByText('File upload limit:')).toBeInTheDocument();
+    expect(screen.getByText('10')).toBeInTheDocument();
```

Comment:
Should these be combined somehow? I just worry that if someone added another number somewhere and it rendered `10` this might pass by finding the other `10` instead of the one referring to the upload limit.